data/pools/transactionPool.go

```diff
@@ -220,17 +245,42 @@ func (pool *TransactionPool) rememberCommit(flush bool) {
 	if flush {
 		pool.pendingTxGroups = pool.rememberedTxGroups
 		pool.pendingTxids = pool.rememberedTxids
+		pool.pendingLastestLocal = pool.rememberedLatestLocal
 		pool.ledger.VerifiedTransactionCache().UpdatePinned(pool.pendingTxids)
 	} else {
+		// update the GroupCounter on all the transaction groups we're going to add.
+		// this would ensure that each transaction group has a unique monotonic GroupCounter
+		encodingBuf := protocol.GetEncodingBuf()
+		for i, txGroup := range pool.rememberedTxGroups {
+			pool.pendingCounter++
+			txGroup.GroupCounter = pool.pendingCounter
+			txGroup.EncodedLength = 0
+			for _, txn := range txGroup.Transactions {
+				encodingBuf = encodingBuf[:0]
+				txGroup.EncodedLength += len(txn.MarshalMsg(encodingBuf))
+			}
+			pool.rememberedTxGroups[i] = txGroup
+			if txGroup.LocallyOriginated {
+				pool.pendingLastestLocal = txGroup.GroupCounter
+			}
+		}
+		protocol.PutEncodingBuf(encodingBuf)
 		pool.pendingTxGroups = append(pool.pendingTxGroups, pool.rememberedTxGroups...)
 
 		for txid, txn := range pool.rememberedTxids {
 			pool.pendingTxids[txid] = txn
 		}
 	}
 
+	pool.resetRememberedTransactionGroups()
```

Comment:
why are we deleting remembered transactions now and didn't do it before?

Reply:
look on the few lines below - we use to reset the remembered transaction groups, but we did not had a function that would do that. I extracted the "resetting remembered transaction groups" into it's own method so that I can call it from few other locations.